File: Apuntes/Nodos/Subcriptor.py
""" 
TODO: Creacion de nodo para poder subscribirse a un topic
"""
#!/usr/bin/env python
import rospy
from std_msgs.msg import Int32

def callback(msg):
    """
    Funcion que se ejecuta cuando se recibe un mensaje
    """
    print ( msg )

rospy.init_node('lectura')#Creamos el nodo
rospy.Subscriber('laser', Int32, callback) #Nos suscribimos al topic laser

rospy.spin() #Mantener el nodo activo

File: Apuntes/C++/Topics/subcriptor.cpp
#include <ros/ros.h>
#include <std_msgs/Int32.h>

void Callback(const std_msgs::Int32::ConstPtr& msg)
{
    /*Funcion que se ejecuta cuando llega un mensaje por el topic subcriptor*/
    ROS_INFO("I heard: [%d]", msg->data);
}

int mian(int argc, char **argv){

    //Creamos el nodo principal 
    ros::init(argc, argv, "subcriptor");
    ros::NodeHandle nh;

    //Creamos el subcriptor
    ros::Subscriber sub = nh.subscribe("topic", 1000, Callback);

    //Creacion del bucle infinito 
    ros::spin();

    return 0;
    
}